### PR TITLE
optimize: auto closeable

### PR DESCRIPTION
**Pull request by jsbxyyx**
### Ⅰ. Describe what this PR did
optimize: auto closeable

### CODE REVIEW

File: core/src/main/java/io/seata/core/store/db/LockStoreDataBaseDAO.java [hunk 1]
```diff
@@ -185,27 +186,7 @@ public boolean acquireLock(List<LockDO> lockDOs) {
...
-            if (ps != null) {
-                try {
-                    ps.close();
-                } catch (SQLException e) {
-                }
-            }
-            if (conn != null) {
-                try {
-                    if (originalAutoCommit) {
-                        conn.setAutoCommit(true);
```
Comment (zjinlei) [inline core/src/main/java/io/seata/core/store/db/LockStoreDataBaseDAO.java]: need to be processed originalAutoCommit

File: common/src/main/java/io/seata/common/util/IOUtil.java [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+ * @author jsbxyyx
+ */
+public class IOUtil {
+
+    /**
+     * close Closeable
+     * @param closeables the closeables
+     */
+    public static void close(AutoCloseable... closeables) {
+        if (closeables != null && closeables.length > 0) {
```
Comment (zjinlei) [inline common/src/main/java/io/seata/common/util/IOUtil.java]: use seata CollectionUtils

**Comment by echooymxq** [comment]
why not use try-with-resources?

**Comment by helloworlde** [comment]
I found all resource in this change could use try-with-resource, and that can be closed automatic. Maybe change to use try-with-resource is better.

**Comment by jsbxyyx** [comment]
@echooymxq @helloworlde Not all places can be used try-with-resources.
image

**Comment by slievrly** [comment]
@jsbxyyx database operation related variable names can be optimized.

**Comment by jsbxyyx** [comment]
@slievrly OK.

**Comment by helloworlde** [comment]
> @echooymxq @helloworlde Not all places can be used try-with-resources.
> image

It's can be closed by multiple try.

**Comment by jsbxyyx** [comment]
> > @echooymxq @helloworlde Not all places can be used try-with-resources.
> > image
> 
> It's can be closed by multiple try.

If use try-with-resources, the `conn` close can not has other logic.

File: common/src/main/java/io/seata/common/util/IOUtil.java [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+/**
+ * @author jsbxyyx
+ */
+public class IOUtil {
+
+    /**
+     * close Closeable
+     * @param closeables the closeables
+     */
+    public static void close(AutoCloseable... closeables) {
```
Comment (xingfudeshi) [inline common/src/main/java/io/seata/common/util/IOUtil.java]: We can merge these two methods into one.
public static void close(AutoCloseable closeable,AutoCloseable... moreCloseables);

**Pull request merged**